packages/vite-plugin-cloudflare/src/index.ts

```diff
@@ -1080,6 +1080,43 @@ if (import.meta.hot) {
 				}
 			},
 		},
+		// Plugin to handle cron/email/etc triggers
+		{
+			name: "vite-plugin-cloudflare:trigger-handlers",
+			enforce: "pre",
+			async configureServer(viteDevServer) {
+				assertIsNotPreview(resolvedPluginConfig);
+
+				const entryWorkerConfig = getEntryWorkerConfig(resolvedPluginConfig);
+				assert(entryWorkerConfig, `No entry Worker config`);
+
+				const entryWorkerName = entryWorkerConfig.name;
+
+				// cron && email triggers
+				viteDevServer.middlewares.use("/cdn-cgi/", (req, res, next) => {
+					const requestHandler = createRequestHandler((request) => {
+						assert(miniflare, `Miniflare not defined`);
+
+						// set the target service that handles these requests
+						// to point to the User Worker (see `getTargetService` fn in
+						// `packages/miniflare/src/workers/core/entry.worker.ts`)
+						request.headers.set(CoreHeaders.ROUTE_OVERRIDE, entryWorkerName);
+						return miniflare.dispatchFetch(request, { redirect: "manual" });
+					});
+
+					// `req.url` is the URL of the request relative to the middleware
+					// mount path. Here we ensure that miniflare receives a request that
+					// reflects the original request url
+					// 🚨🚨🚨 A potential side effect of mutating the original request is
```

Comment:
not sure how concerning this is, but added a comment for posterity explaining why we do this and what possible side effects are. Another option would have been to create a new `IncomingMessage`-like object, but that seemed to be rather complex, so I chose for this option instead. Happy to switch approach though if there are major concerns